src/lib.rs

```diff
@@ -62,17 +67,24 @@ pub type Result<T, E = CoreError> = std::result::Result<T, E>;
 /// expected that only one instance of an implementation will exist for the lifetime of the
 /// worker(s) using it.
 pub trait Core: Send + Sync {
-    /// Ask the core for some work, returning a [Task], which will eventually contain either a
-    /// [protos::coresdk::WfActivation] or an [protos::coresdk::ActivityTask]. It is then the
-    /// language SDK's responsibility to call the appropriate code with the provided inputs.
+    /// Ask the core for some work, returning a [Task], which will contain a [protos::coresdk::WfActivation].
+    /// It is then the language SDK's responsibility to call the appropriate code with the provided inputs.
     ///
     /// TODO: Examples
+    /// TODO: rename to poll_workflow_task and change result type to WfActivation
     fn poll_task(&self, task_queue: &str) -> Result<Task>;
 
+    /// Ask the core for some work, returning a [protos::coresdk::Task], which will contain a [protos::coresdk::ActivityTask].
+    /// It is then the language SDK's responsibility to call the completion API.
+    fn poll_activity_task(&self, task_queue: &str) -> Result<Task>;
```

Comment:
We need some tests for this API -- ideally we can test it in the existing test you made, simulating what the worker would need to do in order for the activity result to actually end up in the history (which would only apply to the incremental version, not the full replay version, like we discussed)

Reply:
The thing is that if we do it with the fake core, these API interactions won't be triggering state machine changes and hence all we'll be testing is that core calls the server on poll/complete. Which is a legitimate use case to test, I just don't think it should be a part of that history replay flow.

Reply:
Added an integ test for it.